Symfony/CS/Tokenizer/Token.php

```diff
@@ -110,14 +118,20 @@ public function equals($other)
     /**
      * Check if token is equals to one of given.
      *
-     * @param array $others array of tokens or token prototypes
+     * @param array      $others        array of tokens or token prototypes
+     * @param bool|array $caseSensitive global case sensitiveness or an array of booleans (for each token)
      *
      * @return bool
      */
-    public function equalsAny(array $others)
+    public function equalsAny(array $others, $caseSensitive = true)
     {
         foreach ($others as $other) {
-            if ($this->equals($other)) {
+            if (is_array($caseSensitive)) {
+                $cs = count($caseSensitive) ? array_shift($caseSensitive) : true;
```

Comment:
I would make that if `$caseSensitive` is an array then it's an associative array, and key is index of $others.
Then if one want to set only index 3 then he won't need to set 0..2 as well.